NETDefault/RequestHandlerNETCore31/RequestHandler.cs

```diff
@@ -0,0 +1,93 @@
+            => HandleResponseAsync(request, logger, requestHandler, functionMethodName);
+
+        /// <summary>
+        /// Invokes requestHandler and catches common exceptions.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="logger"></param>
+        /// <param name="requestHandler"></param>
+        /// <param name="functionMethodName">Gets filled automatically by compiler</param>
+        /// <returns></returns>
+        public static Task<IActionResult> Handle(HttpRequest request, ILogger logger, Func<IActionResult> requestHandler, [CallerMemberName] string functionMethodName = "")
+            => HandleResponseAsync(request, logger, () => Task.FromResult(requestHandler()), functionMethodName);
+
+        /// <summary>
+        /// Validates TRequestModel and invokes requestHandler.
+        /// 
+        /// Catches common exceptions.
+        /// </summary>
+        /// <typeparam name="TRequestModel"></typeparam>
+        /// <param name="request"></param>
+        /// <param name="logger"></param>
+        /// <param name="requestHandler"></param>
+        /// /// <param name="functionMethodName">Gets filled automatically by compiler</param>
+        /// <returns></returns>
+        public static async Task<IActionResult> HandleAsync<TRequestModel>(HttpRequest request, ILogger logger, Func<TRequestModel, Task<IActionResult>> requestHandler, [CallerMemberName] string functionMethodName = "")
+        {
+            var body = await request.ReadAsStringAsync();
+            var requestBody = JsonConvert.DeserializeObject<TRequestModel>(body);
+
+            return request is null
+                ? new BadRequestResult()
+                : await HandleValidationAsync(request, requestBody, logger, requestHandler, functionMethodName);
+        }
+
+        private static async Task<IActionResult> HandleValidationAsync<TRequestModel>(HttpRequest request, TRequestModel requestBody, ILogger logger, Func<TRequestModel, Task<IActionResult>> requestHandler, string functionMethodName)
+        {
+            var validationResult = Utilities.ModelValidator(requestBody);
+            
+            return !validationResult.IsValid
+                ? new BadRequestObjectResult(validationResult.ValidationResults)
+                : await HandleResponseAsync(request, logger, () => requestHandler(requestBody), functionMethodName);
+        }
+
+        private static async Task<IActionResult> HandleResponseAsync(HttpRequest request, ILogger logger, Func<Task<IActionResult>> requestHandler, string functionMethodName)
+        {
+            try
+            {
+                var response = await requestHandler();
+                return response;
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                logger.LogWarning(ex, "{functionName}: Invalid credentials", functionMethodName, request.HttpContext.TraceIdentifier);

```

Comment:
Is unauthorized access a warning?, it could be if you want a easy overview of invalid attempts. Do know that a lot of clients are stupid and will keep retrying until someone replaces expired credentials manually, so this could result in a flood of warnings in the log.

Reply:
Cut it out - not required for a generic example of RequestHandler